File: datacollector/src/main/java/ru/trade/datacollector/messages/Queues.java
package ru.trade.datacollector.messages;

import org.springframework.cloud.stream.annotation.Input;
import org.springframework.cloud.stream.annotation.Output;
import org.springframework.messaging.MessageChannel;
import org.springframework.messaging.SubscribableChannel;
import ru.trade.datacollector.util.QueuesNames;

public interface Queues {

    // Commands
    @Input(QueuesNames.INIT_COMMAND)
    public SubscribableChannel initChainProfitCalcCommand();

    @Output(QueuesNames.EXTSYSTEMCHAINREQ_COMMAND)
    public MessageChannel getChainDataFromExtSystemCommand();

}

File: datacollector/src/main/java/ru/trade/datacollector/messages/Queues.kt
package ru.trade.init.messages

import org.springframework.cloud.stream.annotation.Output
import org.springframework.messaging.MessageChannel
import ru.trade.init.util.QueuesNames

//
// import org.springframework.cloud.stream.annotation.Input
// import org.springframework.cloud.stream.annotation.Output
// import org.springframework.messaging.MessageChannel
// import org.springframework.messaging.SubscribableChannel
// import ru.rit.pkt.messages.QueuesNames
//
interface Queues {
    // Commands
    @Output(QueuesNames.INIT_COMMAND)
    fun initCommand(): MessageChannel
//
//     @Output(QueuesNames.CONNECT_BROADBAND_COMMAND)
//     fun connectBroadbandCommand(): MessageChannel
//
//     // // Events
//     // @Input(QueuesNames.INITIALISE_EVENT)
//     // fun initialiseEvent(): SubscribableChannel
//     //
//     // @Input(QueuesNames.VALIDATE_EVENT)
//     // fun validateEvent(): SubscribableChannel
}